src/libs/antares/study/parts/thermal/cluster_list.cpp

```diff
@@ -712,6 +700,111 @@ void ThermalClusterList::enableMustrunForEveryone()
     each([&](ThermalCluster& cluster) { cluster.mustrun = true; });
 }
 
+void ThermalClusterList::resizeAllTimeseriesNumbers(uint n)
+{
+    assert(n < 200000); // arbitrary number
+    if (not cluster.empty())
+    {
+        if (0 == n)
+        {
+            each([&](ThermalCluster& cluster) { cluster.series->timeseriesNumbers.clear(); });
+        }
+        else
+        {
+            each([&](ThermalCluster& cluster) { cluster.series->timeseriesNumbers.resize(1, n); });
+        }
+    }
+}
+
+bool ThermalClusterList::storeTimeseriesNumbers(Study& study)
+{
+    if (cluster.empty())
+        return true;
+
+    bool ret = true;
+    TSNumbersPredicate predicate;
+
+    each([&](const ThermalCluster& cluster) {
+        study.buffer = study.folderOutput;
+        study.buffer << SEP << "ts-numbers" << SEP << typeID() << SEP << cluster.parentArea->id
+            << SEP << cluster.id() << ".txt";
+        ret = cluster.series->timeseriesNumbers.saveToCSVFile(study.buffer, 0, true, predicate)
+            and ret;
+    });
+    return ret;
+}
```

Comment:
this method can be placed in ClusterList, `typeID()`, `cluster` all make sense there.